core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -568,6 +571,7 @@ class LogManager(logDirs: Array[File],
           log
         } catch {
           case e: IOException =>
+            error(s"Error while creating log for $topicPartition in dir ${dataDir.getAbsolutePath}", e)
```

Comment:
Sure. I have replaced these multiple lines of logging with the logging in `LogDirFailureChannel.maybeAddLogFailureEvent()`. Can you check if it addresses the problem?